backpack-compose/src/main/kotlin/net/skyscanner/backpack/compose/badge/internal/BpkBadgeImpl.kt

```diff
@@ -0,0 +1,158 @@
+        horizontalArrangement = Arrangement.spacedBy(BpkSpacing.Sm),
+        verticalAlignment = Alignment.CenterVertically,
+    ) {
+        val contentColor = type.contentColor
+
+        icon?.let { it() }
+
+        BpkText(
+            text = text,
+            color = contentColor,
+            style = BpkTheme.typography.footnote,
+            maxLines = 1,
+            overflow = TextOverflow.Ellipsis,
+        )
+    }
+}
+
+@Composable
+internal fun BadgeIcon(
+    icon: BpkIcon,
+    size: BpkIconSize,
+    type: BpkBadgeType,
+    modifier: Modifier = Modifier,
+) {
+    BpkIcon(
+        icon = icon,
+        contentDescription = null,
+        size = size,
+        modifier = modifier,
+        tint = type.iconColor,
+    )
+}
+
+@Composable
+internal fun BadgeDrawable(
+    icon: Painter,
+    size: BpkIconSize,
+    type: BpkBadgeType,
+    modifier: Modifier = Modifier,
+) {
+    Icon(
+        painter = icon,
+        contentDescription = null,
+        tint = type.iconColor,
+        modifier = modifier.defaultIconSize(size),
+    )
+}
+
+private val BpkBadgeType.iconColor: Color
+    @Composable
+    get() = when (this) {
+        BpkBadgeType.Warning -> BpkTheme.colors.statusWarningSpot
+        BpkBadgeType.Destructive -> BpkTheme.colors.statusDangerSpot
+        BpkBadgeType.Success -> BpkTheme.colors.statusSuccessSpot
+        else -> {
+            contentColor
+        }
+    }
+
+private val BpkBadgeType.contentColor: Color
+    @Composable
+    get() = when (this) {
+        BpkBadgeType.Normal -> BpkTheme.colors.textPrimary
+        BpkBadgeType.Strong -> BpkTheme.colors.textOnDark
+        BpkBadgeType.Success -> BpkTheme.colors.textPrimary
+        BpkBadgeType.Warning -> BpkTheme.colors.textPrimary
+        BpkBadgeType.Destructive -> BpkTheme.colors.textPrimary
+        BpkBadgeType.Inverse -> BpkTheme.colors.textPrimary
+        BpkBadgeType.Outline -> BpkTheme.colors.textOnDark
+        BpkBadgeType.Brand -> BpkTheme.colors.textPrimaryInverse
+    }
+
+private val BpkBadgeType.backgroundColor: Color
+    @Composable
+    get() = when (this) {
+        BpkBadgeType.Normal -> BpkBadgeColors.backgroundNormal
+        BpkBadgeType.Strong -> BpkTheme.colors.corePrimary
+        BpkBadgeType.Success -> BpkBadgeColors.backgroundNormal
+        BpkBadgeType.Warning -> BpkBadgeColors.backgroundNormal
+        BpkBadgeType.Destructive -> BpkBadgeColors.backgroundNormal
+        BpkBadgeType.Inverse -> BpkTheme.colors.surfaceDefault
+        BpkBadgeType.Outline -> Color.Transparent
+        BpkBadgeType.Brand -> BpkTheme.colors.coreAccent
+    }
+
+private val BpkBadgeType.borderColor: Color
+    @Composable
+    get() = when (this) {
+        BpkBadgeType.Outline -> BpkTheme.colors.textOnDark
+        else -> Color.Transparent
+    }
+
+private val BadgeShape = RoundedCornerShape(BpkBorderRadius.Xs)
+
+private fun Modifier.defaultIconSize(size: BpkIconSize): Modifier =
```

Comment:
i don't think this is needed? badges should only ever have one icon size, so we don't need the `BpkIconSize` prop here - they're always small

Reply:
removed